### PR TITLE
Feature fill wtct with task header

### PR DESCRIPTION
**Pull request by dybi**
resolves: https://github.com/golemfactory/golem/issues/3696
depends on: https://github.com/golemfactory/golem-messages/pull/303

please note, that that: https://github.com/golemfactory/golem-messages/issues/302 in GM should be merged first and requirements must be updated with released version of GM

### CODE REVIEW

**Comment by mfranciszkiewicz** [comment]
Please fix line endings in `golem/task/tasksession.py`

**Comment by dybi** [comment]
@mfranciszkiewicz any reason you have both in the repo? And why not to unify it?

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+--editable git://github.com/golemfactory/golem-messages.git@bd5fed664238070dbd3756a535eb700364daa9f6#egg=Golem_Messages
```
Comment (jiivan) [inline requirements.txt]: Hint: maybe use branch name instead of a single commit?

File: tests/golem/resource/test_resourcehandshake.py [hunk 2]
```diff
@@ -77,19 +77,24 @@ def setUp(self):
...
+            num_cores=10,
+            # task_header=task_header.to_dict(),
         )
         self.session = MockTaskSession(self.tempdir)
         self.session._start_handshake = Mock()
         self.session.task_server.task_keeper.task_headers = task_headers = {}
-        task_headers[self.message['task_id']] = dt_tasks_factory.TaskHeader()
+        task_headers[
+            task_id
+        ] = self.task_header
```
Comment (jiivan) [inline tests/golem/resource/test_resourcehandshake.py]: Very esoteric formatting here :)

File: tests/golem/resource/test_resourcehandshake.py [hunk 3]
```diff
@@ -101,19 +106,22 @@ def test_request_task_handshake(self, *_):
...
+        })
 
         self.session._handshake_required.return_value = False
         self.session.request_task(**self.message)
 
-        assert not self.session.disconnect.called
-        assert self.session.send.called
-
-        slots = self.session.send.call_args[0][0].__slots__
+        self.assertFalse(self.session.disconnect.called)
```
Comment (jiivan) [inline tests/golem/resource/test_resourcehandshake.py]:
```suggestion
        self.session.disconnect.assert_called()
```

File: tests/golem/resource/test_resourcehandshake.py [hunk 4]
```diff
@@ -101,19 +106,22 @@ def test_request_task_handshake(self, *_):
...
+        self.assertTrue(self.session.send.called)
```
Comment (jiivan) [inline tests/golem/resource/test_resourcehandshake.py]:
```suggestion
        self.session.send.assert_called()
```

File: tests/golem/resource/test_resourcehandshake.py [hunk 5]
```diff
@@ -101,19 +106,22 @@ def test_request_task_handshake(self, *_):
...
 
-        msg = message.tasks.WantToComputeTask(**self.message)
-        msg_slots = msg.__slots__
+        sent_wtct = self.session.send.call_args[0][0]
 
-        assert slots == msg_slots
+        msg = message.tasks.WantToComputeTask(**wtct_dict)
+        self.assertEqual(sent_wtct.__slots__, msg.__slots__)
```
Comment (jiivan) [inline tests/golem/resource/test_resourcehandshake.py]: I know it's just a rewrite, but this never made sense. `__slots__` doesn't return slot values, just `__slots__` attribute. 
```suggestion
        self.assertEqual(sent_wtct, msg)
```

File: tests/golem/resource/test_resourcehandshake.py [hunk 6]
```diff
@@ -101,19 +106,22 @@ def test_request_task_handshake(self, *_):
...
+        self.assertEqual(sent_wtct.task_header, self.task_header)
```
Comment (jiivan) [inline tests/golem/resource/test_resourcehandshake.py]:
```suggestion

```

File: tests/golem/task/test_taskmanager.py [hunk 7]
```diff
@@ -408,7 +408,7 @@ def test_computed_task_received(self, _):
             pub_port=1024,
             key="key_id",
         )
-        th = dt_tasks_factory.TaskHeader(
+        th = dt_tasks_factory.TaskHeaderFactory(
```
Comment (shadeofblue) [inline tests/golem/task/test_taskmanager.py:396]: @dybi if we're using the `TaskHeaderFactory`, I don't think we need to hand-generate the Node and provide the task_id and other values here... we can most likely just use the factory as is ... (and possibly, maybe, fix the factory if it doesn't produce a usable `TaskHeader` + `Node` ...

File: requirements.txt (hunk 1 above)
Comment (dybi) [reply requirements.txt]: I will update it with proper tag as soon as GM counterpart is merge and new tag is available

File: tests/golem/resource/test_resourcehandshake.py (hunk 2 above)
Comment (dybi) [reply tests/golem/resource/test_resourcehandshake.py]: no idea why this happened... I'll fix it ;)

File: tests/golem/task/test_taskmanager.py (hunk 7 above)
Comment (dybi) [reply tests/golem/task/test_taskmanager.py:396]: okay, I reduced the boilerplate to the required minimum

File: requirements.txt (hunk 1 above)
Comment (jiivan) [reply requirements.txt]: how about: `-e git://github.com/golemfactory/golem-messages.git@feature-include-task-header-in-want-to-compute-task#egg=Golem_Messages`

**Review by jiivan — APPROVED** [review]
ramon_dekker_middle_vs_middle

**Comment by mfranciszkiewicz** [comment]
@dybi If we change the line endings, it will be difficult to review existing PRs. No other reason, really.

File: tests/golem/task/test_taskmanager.py (hunk 7 above)
Comment (shadeofblue) [reply tests/golem/task/test_taskmanager.py:396]: thank you! :)

**Review by shadeofblue — APPROVED** [review]

**Pull request merged**